packages/tokens/src/fungible/storage.rs

```diff
@@ -83,15 +84,21 @@ impl Base {
     ///
     /// # Notes
     ///
-    /// Attention is required when `live_until_ledger` is less than the current
-    /// ledger number, as this indicates the entry has expired. In such cases,
-    /// the allowance should be treated as `0`.
+    /// For SAC compatibility, we do not return an error for allowances that do
+    /// not exist or have expired.
     pub fn allowance_data(e: &Env, owner: &Address, spender: &Address) -> AllowanceData {
         let key = AllowanceKey { owner: owner.clone(), spender: spender.clone() };
-        e.storage()
+        let allowance_data = e
+            .storage()
             .temporary()
             .get(&StorageKey::Allowance(key))
-            .unwrap_or(AllowanceData { amount: 0, live_until_ledger: 0 })
+            .unwrap_or(AllowanceData { amount: 0, live_until_ledger: 0 });
+
+        if allowance_data.live_until_ledger < e.ledger().sequence() {
+            AllowanceData { amount: 0, live_until_ledger: allowance_data.live_until_ledger }
```

Comment:
Wouldn't it make more sense to return default (0, 0) when expired?

This is in temporary storage and its TTL is set to `live_until_ledger`. It can be changed later and TTL can be extended but again up to the `live_until_ledger`. After `live_until_ledger` the entry gets deleted and the only scenario where the entry can outlive `live_until_ledger` will be when someone bumps it manually which doesn't make sense and should be discouraged. By following this reasoning, I think it's fine to return the default when it's expired.